src/js/commands/format-block.js

```diff
@@ -1,36 +1,24 @@
 import TextFormatCommand from './text-format';
-import { getSelectionBlockElement, selectNode } from '../utils/selection-utils';
-import { inherit } from 'content-kit-utils';
 
-function FormatBlockCommand(options) {
-  options = options || {};
-  options.action = 'formatBlock';
-  TextFormatCommand.call(this, options);
-}
-inherit(FormatBlockCommand, TextFormatCommand);
-
-FormatBlockCommand.prototype.exec = function() {
-  var tag = this.tag;
-  // Brackets neccessary for certain browsers
-  var value =  '<' + tag + '>';
-  var blockElement = getSelectionBlockElement();
-  // Allow block commands to be toggled back to a text block
-  if(tag === blockElement.tagName.toLowerCase()) {
-    throw new Error('Unimplemented: Type.BOLD.paragraph must be replaced');
-    /*
-    value = Type.PARAGRAPH.tag;
-    */
-  } else {
-    // Flattens the selection before applying the block format.
-    // Otherwise, undesirable nested blocks can occur.
-    // TODO: would love to be able to remove this
-    var flatNode = document.createTextNode(blockElement.textContent);
-    blockElement.parentNode.insertBefore(flatNode, blockElement);
-    blockElement.parentNode.removeChild(blockElement);
-    selectNode(flatNode);
+class FormatBlockCommand extends TextFormatCommand {
+  constructor(options={}) {
+    super(options);
   }
 
-  FormatBlockCommand._super.prototype.exec.call(this, value);
-};
+  exec() {
+    const editor = this.editor;
+    const activeSections = editor.activeSections;
+
+    activeSections.forEach(s => {
+      editor.resetSectionMarkers(s);
+      editor.setSectionTagName(s, this.tag);
```

Comment:
I've been thinking of the editor model as the keeper of the public api, and these commands as plugin-style consumers of that public api. It has made me reluctant to have them call methods on the constituent objects directly. Delegating methods through the editor allows it to also take care of its own render-related bookkeeping. I can see the utility of considering methods like `section#setTagName` (or similar) as part of the public API, but it seems like they may cause the sections and markers to end up more tightly coupled to the render tree as a result.